Destroy/qdel

PsiOmegaDelta opened the pull request:
Continues the painful process of porting Destroy()s, Dels(), and everything else needed for qdel.
Debugging using mass singularity/Nar-Sie destruction. Thanks Nar-Sie

Fixes #8870.

PsiOmegaDelta committed:
Drops the machinery part of the singularity.

PsiOmegaDelta committed:
Switches around turf/datum garbage handling (my fault..). Adds null check for zone rebuilds.

PsiOmegaDelta committed:
Cleans up the Status panel.

PsiOmegaDelta committed:
The Nar-Sie summoning ritual is now animated.

PsiOmegaDelta committed:
Continues the painful process of porting Destroy()s, Dels(), and everything else needed for qdel.

PsiOmegaDelta committed:
Mecha + misc. Destroy() because I had already begun.

PsiOmegaDelta committed:
Updates the atom_pool, now datum_pool, to handle any datum object.

RemieRichards commented on code/__HELPERS/datum_pool.dm:
```diff
@@ -50,38 +47,50 @@ var/global/list/GlobalPool = list()
+	var/datum/D = pick_n_take(GlobalPool[get_type])
+	if(D)
+		D.ResetVars()
+		D.Prepare()
```
This takes arguments, you didn't give it any.

PsiOmegaDelta committed:
Updates the atom_pool, now datum_pool, to handle any datum object.

Makes the garbage collector similarly robust. Continues the whole Destroy/qdel porting.

PsiOmegaDelta commented:
Can't even get a mere /obj to garbage collect properly. Moments prior to manual deletion the reference check returns:
````
## TESTING: Beginning search for references to a /obj.
## TESTING: Collected list of things in search for references to a /obj. (519025 Things)
## TESTING: Found /obj [0x20045ff] in /area/turret_protected/ai's contents list var.
## TESTING: Found /obj [0x20045ff] in /turf/simulated/floor/bluegrid's contents list var.
## TESTING: Completed search for references to a /obj.
````
Still get ````## TESTING: GC: -- [0x20045ff] | /obj was unable to be GC'd and was deleted --````.
Will try to make the GC call the reference checker automatically on such issues,  doubt it'll help much.

PsiOmegaDelta committed:
Adds Destroys.

PsiOmegaDelta committed:
Merge remote-tracking branch 'upstream/dev' into TheSummoning

Conflicts:
	code/game/machinery/cryopod.dm
	code/game/objects/items/weapons/clown_items.dm

PsiOmegaDelta commented:
Making multiple PRs for this. Server now starts cleanly without reporting objects failing to be GCd.
Not even the 880+ objects waiting to be destroyed at start of round appear to cause any performance issues.
Use of pooling is included, pipenet improvements will be separate.
